service/service_validate.go

```diff
@@ -34,6 +34,9 @@ func ValidateService(service *Service) error {
 			errs = append(errs, fmt.Errorf("%s%s", namespace, e.Translate(translator)))
 		}
 	}
+	if errs.ErrorOrNil() != nil {
```

Comment:
why this pre-exit?
The rest of the code is appending the errors so we don't need to exit here, that sounds like an early optimization or (and I think it's the case) to exit when the struct has an error (like config missing).
In that case maybe this function should only return an error and exit on the first error that would make things way easier

Reply:
because the rest of the tests will panic as they access pointer that could be nil.
Check line 46: https://github.com/mesg-foundation/engine/pull/1112/files/70e9a6664e42aeab17e1687f2357506ea0d2a76c#diff-fdf7d56c1eb9f4c5fa9f1276efde04e9R46

Reply:
This function is actually doing too many things:
1. check the integrity of the struct
2. check stuff on the data of the struct

if #1 has error, it should return them. that what this line is doing

Reply:
I splitted the function `ValidateService` in 2 functions, one to check the struct integrity, the other one to check the data.